### PR TITLE
Adds new station map Nebulastation

### PR DESCRIPTION
**Pull request by KikoWen0**
## About The Pull Request

Adds a completely brand new multi-z station map "NebulaStation". 
New areas and few exclusive map shuttles (ferry, mining, arrival, evac, cargo)
Few example screenshots
StrongDMM-2024-07-19 20 53 15
StrongDMM-2024-07-19 20 51 41
image
image
image
image
And more...

## Why It's Good For The Game

New maps are always good for the game

## Changelog

:cl: Kiko
add: Nebulastation, new station map
/:cl:

### CODE REVIEW

**Comment by Singul0** [comment]
Looks great! love the style and aesthetics, could easily pass as a syndie or derelict station

**Comment by SomeoneYouProbablyKnow** [comment]
Feels like a very "lived in" station

**Comment by PhantornRU** [comment]
Cozy and awesome station. But you went too far with the decorations on the AI ​​satellite.

**Comment by Rex9001** [comment]
Looks dope, I enjoy the cozyness and detail of the map

**Comment by SmArtKar** [comment]
Attention to detail is incredible, also absolutely love the plantlife onboard. This is like Kilo if it was bigger, better and more modern!

**Comment by DrDiasyl** [comment]
Love that multi-z is used to expand departments and not just stack them up on each other.
There is almost no hallways on 2 floor, very cool in my opinion!

**Comment by Eronymun** [comment]
Good luck Kiko, rooting for you.

I've been watching this map develop for a while, it has a level for attention to detail that no other map we currently have in rotation does. The amount of decoration is staggering. If birdshot is the poor station, then I think this fits the niche as the rich station.

Also someone help Kiko pass Linters, it can be very confusing for a first time PRer.

**Comment by mc-oofert** [comment]
personally the amount of visual clutter doesnt sit well with me but i do think this is an otherwise good station

**Review by MrMelbert — CHANGES_REQUESTED** [review]
I believe every entire upper z-level space turf is using the wrong type of space (`/turf/open/space/basic` instead of `/turf/open/space/openspace`)

This means a considerable amount of the station doesn't have a roof?

image

image

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Also since I have the map open

<Details>

<Summary>

It's become general mapping practice to have maint own all of its walls, which doesn't seem to be the case in much of the map 

</Summary>

image


</Details>

**Review by MrMelbert — COMMENTED** [review]
I saw this so you don't have to (HoS office) 

image

**Review by MrMelbert — COMMENTED** [review]
Virology cyclelink helpers are backwards

image

**Review by MrMelbert — COMMENTED** [review]
<Details>

<Summary>

Make sure you do a fire alarm pass. 
I'm noticing many sections of the map (examples provided), some rather large, which could very easily trap someone with firedoors, as there's no fire alarms anywhere nearby or within. 

</Summary>

image

image

image

image

</Details>

Generally if you have any size area which can have all exits enclosed via firedoor, it should either have an alarm, or an exit without a firedoor which leads somewhere safe (maint).

**Review by MrMelbert — COMMENTED** [review]
I haven't seen very many but make sure you're not stacking wall mounts like this, it makes life difficult for AIs

image

**Review by MrMelbert — COMMENTED** [review]
1. I think in general we'd prefer if we didn't have the roundstart SM be "pre-upgraded" (IE, the volume pumps should just be gas pumps) 
2. The scrubber pipes should be green and the vent pipes should be cyan. So people don't get confused
3. Why are there railings around the SM? Hugbox, remove them

image

**Comment by MrMelbert** [comment]
Third spare in the bank vault should probably not be there 

image

But also, you don't seem to have a spare ID safe on the map

image

**Comment by MrMelbert** [comment]
Leaving it there for now, overall very, very nice looking map. I like the layout and the decal work is spectacular, though I worry that its intricate designs will turn off other contributors from helping in maintaining it.

You should ask in the mapping channel if you need help fixing lints

**Comment by TheVekter** [comment]
We're so fucking back boys

**Comment by Cheshify** [comment]
Huh. Neat.

**Comment by Inari-Whitebear** [comment]
It's fancy and all, but I hate the plasmaglass flooring above the SM chamber, as it can actually just burn up in my experience on Wawa

**Comment by MrMelbert** [comment]
That sounds more like a bug with plasmaglass floors

**Comment by CliffracerX** [comment]
This goes *hard*, I can't wait to see it in action.

**Comment by Blacklist897** [comment]
Baller, this looks great

**Comment by zxaber** [comment]
I absolutely love the supermatter room.

Robotics looks a tad cramped; the two exofabs facing eachother is a neat idea but will lead to the two robos basically being in eachother's space constantly. I wish there was a way to get like one extra tile of width there. I do like the medical corner of Robotics being sectioned off while still within view, though.

**Comment by KikoWen0** [comment]
@zxaber 

Is this design better?
image

**Comment by zxaber** [comment]
It's better. The biggest thing is having somewhere to place "in-progress projects". Leaving your mech chassis at the print location is a pain becuase you have to sift through the parts to click on the chassis. Same for making borgs. And if you're printing out extra things (mech weapons, cells, a posi cube) it just gets worse. On most maps, I'd move them just one tile away (and stand to the side, giving me access to the exosuit fab, print output tile, and project all at once).

If you can find somewhere else for the circuit imprinter (and swap the quad with the borg endoskeleton), you could move the railings up a tile and it would make the area much more open.

**Comment by KikoWen0** [comment]
@zxaber

image

**Comment by zxaber** [comment]
Looks great.

**Review by spockye — CHANGES_REQUESTED** [review]
<details>
<summary>Review</summary>
<br>

image

weird cabling. intentional?

image

same here

image

missing cabling

image

missing cabling

image

missing cabling

image
piping error

image

wild decal!

-------
SECOND Z
---

image

wild cable

image

wild cabling

image

wild cabling

image

wild atmos

image

wild atmos

</details>

--------

This is really really really pretty. AMAZING work Kiko <3
i can tell you've put a shit ton of work into this!

**Comment by GremlinSeeker** [comment]
looks gorgeous from the screenshots, very well done.

**Comment by TheVekter** [comment]
Going to run a round or two of this tonight on Manuel if anyone is interested.

**Comment by zombiegums** [comment]
This station is beautiful. I love it, especially the bar/kitchen and pool area.

**Comment by KingkumaArt** [comment]
ok so I was browsing the current pull requests, because I have no life, and saw "new map" with "good first pr" on it, and was a bit worried. Mapping is hard.

I havent been happier to be wrong on a pr ever. This map is... it genuinely may be my favorite map we have. It feels like a place that could actually exist - akin to kilo or bird - items laid out on tables randomly, lazily, not in neat rows like they were placed by a machine - deserves special mention for the "lived-in" feeling.  it's also multi-z without having one or more of the zlevels feeling half-empty, which often happens (icebox and wawa come to mind). There's other stuff - the cargo containers in cargo, the plasma glass floor in engineering, the plants everywhere - but i want to be SOMEWHAT brief.

Somehow, (and I know this is gonna sound insane), it feels like if ss13 was an actual place, it would be like this one. To me, it looks like it was just a small space outpost that continually got remodeled and had extensions bolted on into some Space Winchester Mystery House fits the "nonsensical, metal death trap masquerading as a space station" incredibly well. 

Genuinely incredible. 10/10.

**Comment by MrMelbert** [comment]
Disposals mass driver controller is not connected to the mass driver 

image

**Review by MrMelbert — CHANGES_REQUESTED** [review]
<Details>
<Summary>
Lots of active turfs. Spawn in game and turn on the markers (1059 markers placed)

</Summary>

image

image

</Details>

**Comment by MrMelbert** [comment]
These signs cover up the elevator panel when raised

image

**Review by MrMelbert — COMMENTED** [review]
<Details>

<Summary>
Intercom dead zone in xenobio
</Summary>

image

</Details>

**Comment by MrMelbert** [comment]
I love the idea behind this locker, but some of the guns and armor in this locker are tooooo power-creepy

- Swat boots are a big NO, they're meant for deathsquad members (they have really good armor and are no-slips
- Auto-rifle is meant to be from cargo only
- Fischer is meant to be from black market only 
- The Marine armor is also supposed to be ERT only, they boast pretty good armor and space protection 

image

**Comment by MrMelbert** [comment]
- Having the only way into the lower bridge (besides maint) be an elevator is a fire hazard. smh

**Comment by MrMelbert** [comment]
\- The pool tiles are named "beach". Maybe want to make a subtype for it and rename it?

image

**Comment by MrMelbert** [comment]
\- Unconnected vent

image

**Comment by MrMelbert** [comment]
\- No mass driver controller in ord

image

**Comment by MrMelbert** [comment]
\- Hos's privacy shutters require QM access
\- Cap's privacy shutters require RD access
\- CE's privacy shutters require no access

**Comment by MrMelbert** [comment]
\- Incinerator supply tank appears to be setup incorrectly.

**Comment by MrMelbert** [comment]
\- This button doesn't seem to do anything

image

**Comment by MrMelbert** [comment]
\- Medical's closet recharge station to the break room is all the way down stairs in the break room bathroom, which may be rather inconvenient for cyborgs and modsuit users. Maybe add one in the cryo room?

**Comment by MrMelbert** [comment]
\- Having map-ready operating setups directly in the treatment center is a tad power-creep, we usually expect the players to set it up if desired or otherwise use the surgical theatres. I'd suggest you remove them (but I don't know what you'd fill the space with)

image

**Comment by MrMelbert** [comment]
\- You don't have any `/obj/structure/sign/painting/library_secure` mapped in, just private and public

**Comment by MrMelbert** [comment]
Nittiest of nitpicks but I think TerraGov's and the SSC's flag should be swapped.  The station does canonically exist in SSC space and I imagine their representatives would petition to be put next to the NT flag, front and centre. 

image

**Comment by MrMelbert** [comment]
\- None of these windoors have access set, even though the generators inside require teleporter access to use.

image

**Comment by MrMelbert** [comment]
Decal moment

image

**Comment by MrMelbert** [comment]
These shutters in genetics have no button (also make sure to give the button access)
image

**Comment by MrMelbert** [comment]
These doors are named after an escape pod, but there's no pod here. 
Also their one-way-access is the wrong way. 
Also also they require security access instead of external airlock.

image

**Comment by MrMelbert** [comment]
Some of the dirt turfs you use around the station baseturf to chasms 
Which means if they get destroyed, you have chasms on the station 
This is fine for things on the upper level (they fall down 1 z-level) but for things on the lower level - it kills people

image

**Comment by KikoWen0** [comment]
Thanks @MrMelbert I'll fix everything in an hour or so

**Comment by Blacklist897** [comment]
the marine armor in armory has a damaged subtype that looks the same but has less armor and no space proofing if you have to replace it
engineering and sci dont start with flatpackers
Apart from that, this map has this amazing lived in feel and the single main hallway is a really cool layout

**Comment by Pumpkinoe** [comment]
really really good map design, but there's a lot of turf decals around the map that feel like there's wayyy too many decals in one spot (or just really messy)
a lot of it doesn't communicate much and sometimes it's trying to communicate too many things at once

**Comment by Pumpkinoe** [comment]
> a lot of it doesn't communicate much and sometimes it's trying to communicate too many things at once

an example of this can be seen in how many of the circle trimlines have been used around the station - like departures has the 'box'/circle trimline inbetween the benches withtout it really needing to be there outside of filling up detail space

it looks pretty but it's also kinda messy 

while the 'box' trimlines are likely the worst offenders of it, it also happens in places like xenobio which has trimline, classic tile decals, and even caution lines underneath it all, all in the same place

there's a ton of really, *really* good design choices with decals - like the trimlines being used to suggest levels of distance in the security firing range, or the cute mural in the fishing den - but i think a ton of it ends up being very visually messy and hard to use for visual information

**Comment by MoffNyan** [comment]
I love Nebulastation!! it's beautiful, I especially love how you decorated the floor tiling through the halls. It's absolutely clear that love and care went into every single tile. It does feel a bit visually overstimulating since there's not a lot of "White Space" to give the viewers eyes a chance to rest, the entire station feels very cramped and difficult to navigate - compared to many of the other stations which are streamlined and very simple which makes it easier for your brain to process.
But I don't think that's a bad thing. It's a nice change of pace from the other stations which are pretty simple! Over time it'll become easier to remember all the routes and areas, just like with any other map, it just may take longer due to the complexity and size of the map. I think it would be a shame to revert or simplify such a beauty of a map from this point! I do feel really inspired by the way you designed the map, and I think I've got a lot to learn from your designs too! There's plenty of space for gimmicks or redesigning different spots, lots of materials around, and it's just a generally pleasant station to be in.

I did find one exploitable oversight, when I was exploring maints on the upper Z level, I fell into this elevator connected to security, I couldn't exit or take the elevator back up, BUT, the security doors weren't given the correct permissions, allowing anyone (or at least an assistant) to walk through the door downwards or to the right. All the other doors were inaccessible. Though it did give me access to the Security protolathe and then the disposals chute to get me out of there.

SecurityElevator

I also forgot to take a picture of the janitor's closet, but the disposals sends you to the custodial closet which you can't leave from and there's no way for outsiders to see that you're stuck in there. Janitors almost never go back to the custodial closet except at roundstart to get items, so I feel like this could become a bit of an issue. Usually the chutes send you to Cargo, somewhere that's usually pretty easy to spot, and there's always a lot of traffic around cargo too. I feel bad for any poor Mime that can't ask for help to get out of the custodial closet and won't be seen.
Of course, you can always try to break/hack the door or windows to get out...

**Comment by MoonDeer728** [comment]
not the z level noooo

**Comment by WarerBrow** [comment]
> not the z level noooo

yeah until we fix that [z-level bug](https://github.com/tgstation/tgstation/issues/82804) all maps with z are kinda meh to me

**Comment by iansdoor** [comment]
Wow that was a fantastic mapping. 

A bunch of missing wires on apcs. 
AI sat
Screenshot_699
your laser tag arena doesn't have wire to power the room!!
Screenshot_698

**Comment by OuroborusReporter** [comment]
A large chunk of maint between atmospherics and security on Nebulastation seems to not have a corresponding APC - The entire section was fully unpowered as a result, allowing you to do fun things like break into the back rooms of security with just a crowbar.

Screenshot 2024-07-12 140000

**Comment by necromanceranne** [comment]
I've enjoyed quite a few different maps over time, but I think this one takes the cake as my 100% favourite map, without a hint of exaggeration. The attention to detail and the general layout is really fun. While the directional markers for certain parts of the station are a little confusing at times, once you figure out where generally you are, it tends to be a little easier. I knew exactly where I was just by looking at how colour has been used for the area.

I really like that the beautification of the station is both practical for triangulation and visually interesting, and keeps the station from feeling like a sequence of grey boxes, which is a major problem for many of the other maps. That prior design hinders your ability to navigate in some ways, as many locations are more or less indistinguishable. I never quite felt like I was lost on this map even though it is a lot to take in at first.

I think you've set a new standard for our mappers that might be hard to beat.

**Comment by bob-b-b** [comment]
Enjoyed the rounds I've played so far even if the room shapes and movement flow are a bit wacky compared to our other maps.

In-terms of feedback: ordnance seems to only have three tank transfer valves which is significantly less than other maps, 
the binoculars in ordnance launch room are almost perfectly hidden under the range camera screen, 
also as someone mentioned already science and engineering don't have flatpacked flatpackers ready for deployment.

**Comment by Admiral-Gigaton** [comment]
> Nittiest of nitpicks but I think TerraGov's and the SSC's flag should be swapped. The station does canonically exist in SSC space (iirc) and I imagine their representatives would petition to be put next to the NT flag, front and centre.
> 
> image

NT is canonically in french territory?

**Comment by necromanceranne** [comment]
Apparently the ordnance burn chamber igniter doesn't start connected to the activation device.

**Comment by KikoWen0** [comment]
@necromanceranne I did put some unres. access for robos to access the morgue, but I don't really know why igniter does not work. It is, in fact, connected with the button, but it turns on for 1-2 seconds and does nothing, not really sure if it's the map issue because I did exactly the same setup as all maps do.

**Comment by KikoWen0** [comment]
Same goes to mass driver at disposals, it should be connected, but it isn't for some reason and I don't know how to fix this issue, because the setup is identical to other maps.

**Comment by Mothblocks** [comment]
@tgstation/maptainers You need to comment on this

**Comment by zombiegums** [comment]
Maintenance doesn't appear to be safe from radstorms. Was in medical maints (just below cryo) and my patient was getting irradiated.

Encountered this issue here:
https://statbus.space/rounds/232147

**Comment by norsvenska** [comment]
psychology office has two fireplaces stacked on one another

**Comment by DaydreamIQ** [comment]
Absolutely adore how expansive perma Is instead of a box with nothing in it. Bit odd that the kitchen isn't stocked with milk but you can just make it so eh

**Comment by JupiterJaeden** [comment]
Just a general comment/feedback: the station could use a lot more red & blue emergency lockers, fire extinguisher cabinets, and newscasters. It's currently way behind most stations in terms of the number of these.

**Comment by deathrobotpunch** [comment]
I love the map though I think cargo is a bit too small

**Comment by Twaticus** [comment]
im late to the nebula party but your attention to detail is impeccable and im thoroughly impressed

my personal gripe with this is i love the sandbox nature of the game so i prefer more 'plain' blank slate maps that people can doll-up freely during a round, and maps that are TOO good like this kinda make you not want to make changes cause its already perfect. doesnt matter just my two cents either way youre crazy talented nice job

**Comment by optimumtact** [comment]
OP said this project was suspended so closing the PR for now.

**Comment by ninjanomnom** [comment]
Author requested reopen

**Comment by deathrobotpunch** [comment]
hooray!

**Comment by Loafin34** [comment]
I feel like you made a rule where there has to be at least 2 hedges per room. it makes it feel a bit samey.

**Review by Wallemations — APPROVED** [review]
the only sprite changes are more mapping area sprites so I'm just approving to get the notif out of the way lmao

**Comment by Likteer** [comment]
Very pretty to look at, but impractical.
Like, the map is enormous yet very claustrophobic in populated spaces.
Sprawling, with many locations spelling permadeath for anyone who succumbs there.

Feels more like a heaven for specific antagonists than a livable place.
Still, very pretty.
<details><summary>...</summary>
<p>

Kilo was also so pretty :(

</p>
</details>

**Comment by TheVekter** [comment]
There's currently an issue with people using camera consoles becoming stuck and unable to stop interacting with them. The map is being removed from testmerge until this can be resolved.

**Comment by necromanceranne** [comment]
That isn't a map issue, surely? Wouldn't that be a problem with the consoles?

**Comment by Aliceee2ch** [comment]
image
thats goes hard, fix pleeease c:

**Comment by iliyaxox** [comment]
hai. i was gonna make a pr about this but didn't know this was a tm'd map. great map btw, hope it gets merged

all autodrobes (except for one) are failing to spawn because of the `/all_access` at the end of their type. after checking, autodrobes were supposed to be in these spots:

image

image

image

File: _maps/nebulastation.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"traits": [
+		{
+			"Up": true,
+			"Linkage": "Cross"
+		},
+		{
+			"Down": true,
+			"Baseturf": "/turf/open/openspace",
+			"Linkage": "Cross"
+		}
```
Comment (timothymtorres) [inline _maps/nebulastation.json:21]:
```suggestion
	"traits": [
		{
			"Linkage": "Cross"
		},
		{
			"Baseturf": "/turf/open/openspace",
			"Linkage": "Cross"
		}
```

Comment (timothymtorres) [reply _maps/nebulastation.json:21]: In #87029 multi-z autolinks the up/down traits without having to do it via JSON.

**Comment by PowerfulBacon** [comment]
I don't play TG but I have to say, this map looks absolutely gorgeous. Well done

**Comment by tgstation-server** [comment]
#### Test Merge Deployed

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\vekter

##### Revision
Origin: cded23b51a2264d636a336bd645f4714bdb1e82e
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: ea5e2c43cb2f0b0b8296a483634cb9ce36c89de1
[GitHub Deployments](https://github.com/tgstation/tgstation/deployments/activity_log?environment=TGS%3A+Terry)
</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\vekter

##### Revision
Origin: 02751843692c704de9ffb1fdd98b83b45d22c147
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 77be315d105c033cdd852c26dfcb4d881ca52dfa
</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\vekter

##### Revision
Origin: 6f2267b624373846de249554931e034271786ef5
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 9f1c7eafc622645a3e6ecd162d1b033906a7022a
</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\vekter

##### Revision
Origin: 95cc2588457cdc637c6d7a9a5d12997016ae3d71
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 98ae8f6d5dad954850a84a4ccddffd102c599e40
</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\mrstonedone

##### Revision
Origin: f94ed49e023a6bfa341f9f88b8b8eecb7ffa35e6
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 3164bd3fa801861c04a00130338e7d4beb27a59c

##### Comment

[GitHub Deployments](https://github.com/tgstation/tgstation/deployments/activity_log?environment=TGS%3A+Terry)
</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\mrstonedone

##### Revision
Origin: d9350cf31df20d0829d68d07a7bf30cfef514e96
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 8faa83d9b1bff81932a5afb5cb7d7eb8ccd40053

##### Comment

</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\mrstonedone

##### Revision
Origin: 06aa1279d4a3543fd71620db610a96c585c2bfee
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 6acadb30f09bd049cd04e0d3cbc784a3ea55ee91

##### Comment

</details>

**Comment by tgstation-server** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
Terry

##### Merged By
Auto\time-green

##### Revision
Origin: 06aa1279d4a3543fd71620db610a96c585c2bfee
Pull Request: becfb2aeced38269debf6eba914f2a158a481dab
Server: 4ebdc5e4e6f7ee9eb599ebbb1ddce99adf8a228e

##### Comment

[GitHub Deployments](https://github.com/tgstation/tgstation/deployments/activity_log?environment=TGS%3A+Terry)
</details>

**Comment by LemonInTheDark** [comment]
fucking bot

**Comment by DaydreamIQ** [comment]
The Doppler array thing for toxins is rotated the wrong way and won't detect test site explosions

**Comment by KikoWen0** [comment]
> The Doppler array thing for toxins is rotated the wrong way and won't detect test site explosions

@DaydreamIQ Should be fixed

**Pull request merged**